src/components/modules/readonly.ts

```diff
@@ -0,0 +1,104 @@
+import Module from '../__module';
+import * as _ from '../utils';
+
+/**
+ * @module ReadOnly
+ *
+ * Has one important method:
+ *    - {Function} toggleReadonly - Set read-only mode or toggle current state
+ *
+ * @version 1.0.0
+ *
+ * @typedef {ReadOnly} ReadOnly
+ * @property {boolean} readOnlyEnabled - read-only state
+ */
+export default class ReadOnly extends Module {
+  /**
+   * Value to track read-only state
+   *
+   * @type {boolean}
+   */
+  private readOnlyEnabled = false;
+
+  /**
+   * Returns state of read only mode
+   */
+  public get isEnabled(): boolean {
+    return this.readOnlyEnabled;
+  }
+
+  /**
+   * Set initial state
+   */
+  public async prepare(): Promise<void> {
+    if (!this.isToolsReadOnlyCompatible()) {
+      return;
+    }
+
+    this.readOnlyEnabled = this.config.readOnly;
+  }
+
+  /**
+   * Set read-only mode or toggle current state
+   * Call all Modules `toggleReadOnly` method and re-render Editor
+   *
+   * @param {boolean} state - (optional) read-only state or toggle
+   */
+  public async toggle(state = !this.readOnlyEnabled): Promise<boolean> {
+    if (!this.isToolsReadOnlyCompatible()) {
+      return false;
+    }
+
+    this.readOnlyEnabled = state;
+
+    for (const name in this.Editor) {
+      /**
+       * Verify module has method `toggleReadOnly` method
+       */
+      if (!this.Editor[name].toggleReadOnly) {
+        continue;
+      }
+
+      /**
+       * set or toggle read-only state
+       */
+      this.Editor[name].toggleReadOnly(state);
+    }
+
+    /**
+     * Save current Editor Blocks and render again
+     */
+    const savedBlocks = await this.Editor.Saver.save();
+
+    await this.Editor.BlockManager.clear();
+    await this.Editor.Renderer.render(savedBlocks.blocks);
+
+    return this.readOnlyEnabled;
+  }
+
+  /**
+   * Return true if all tools support read-only mode
+   */
+  private isToolsReadOnlyCompatible(): boolean {
+    const { Tools } = this.Editor;
+    const { blockTools } = Tools;
+    const toolsDontSupportReadOnly: string[] = [];
+
+    Object.entries(blockTools).forEach(([name, tool]) => {
+      if (!Tools.isReadOnlySupported(tool)) {
+        toolsDontSupportReadOnly.push(name);
+      }
+    });
+
+    if (toolsDontSupportReadOnly.length > 0) {
+      _.logLabeled(
```

Comment:
for now, this log is displaying event if I do only initializing editor in a regular mode, on the first load.

It should be displayed only when initializing in read-only mode or when using 'toggle' method.